### PR TITLE
fix(@angular-devkit/build-angular): fix issue were `@media all` causing critical CSS inling to fail

### LINKED ISSUE #1105: Is there a way to compress all the css and all html template and load it in one shot in production mode, instead of making multiple calls
> Please provide us with the following information:
> ---------------------------------------------------------------
1. OS?  Windows 7, 8 or 10. Linux (which distribution). Mac OSX (Yosemite? El Capitan?)

2. Versions. Please run `ng --version`. If there's nothing outputted, please
   run in a Terminal: 
        node --version
   And paste the result here.

3. Repro steps. Was this an app that wasn't created using the CLI? What change did you
   do on your code? etc.

4. The log given by the failure. Normally this include a stack trace and some
   more information.

5. Mention any other details that might be useful.
> ---------------------------------------------------------------
> > Thanks! We'll be in touch soon.

### LINKED ISSUE #20804: Index html generation failed on ng build in Angular 12
# Bug Report

### Affected Package

Not sure which package is causing this error, here are the packages in use
```
{
  "name": "quiz-app",
  "version": "0.0.0",
  "scripts": {
    "ng": "ng",
    "start": "node server.js",
    "build": "ng build",
    "watch": "ng build --watch --configuration development",
    "test": "ng test"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "~12.0.0",
    "@angular/cdk": "^12.0.0",
    "@angular/common": "~12.0.0",
    "@angular/compiler": "~12.0.0",
    "@angular/core": "~12.0.0",
    "@angular/flex-layout": "^11.0.0-beta.33",
    "@angular/forms": "~12.0.0",
    "@angular/material": "^12.0.0",
    "@angular/platform-browser": "~12.0.0",
    "@angular/platform-browser-dynamic": "~12.0.0",
    "@angular/router": "~12.0.0",
    "@types/prismjs": "^1.16.5",
    "express": "^4.17.1",
    "lodash": "^4.17.21",
    "moment": "^2.29.1",
    "ngx-cookie-service": "^11.0.2",
    "path": "^0.12.7",
    "perfect-scrollbar": "^1.5.1",
    "prismjs": "^1.23.0",
    "rxjs": "~6.6.0",
    "tslib": "^2.1.0",
    "zone.js": "~0.11.4"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "~12.0.0",
    "@angular/cli": "~12.0.0",
    "@angular/compiler-cli": "~12.0.0",
    "@types/jasmine": "~3.6.0",
    "@types/lodash": "^4.14.169",
    "@types/node": "^12.11.1",
    "jasmine-core": "~3.7.0",
    "karma": "~6.3.0",
    "karma-chrome-launcher": "~3.1.0",
    "karma-coverage": "~2.0.3",
    "karma-jasmine": "~4.0.0",
    "karma-jasmine-html-reporter": "^1.5.0",
    "typescript": "~4.2.3"
  },
  "engines": {
    "node": "14.17.0",
    "npm": "6.14.13"
  }
}
```

### Description
While running the build using `ng build` command, the build fails to generate the Index HTML file with the following error
Screenshot 2021-05-15 at 12 46 41 PM

## Minimal Reproduction

Check the build log: https://github.com/anuj9196/quiz-app/runs/2589355739?check_suite_focus=true#step:7:56

To reproduce
* take a clone of the repository
* Install dependencies `npm install`
* Build the project `ng build`

## Exception or Error
<pre><code>
✔ Browser application bundle generation complete.
✔ Copying assets complete.
✖ Index html generation failed.
undefined:4:720308: missing '}'
</code></pre>

## Your Environment

**Angular Version:**
<pre><code>
Angular CLI: 12.0.0
Node: 14.17.0
Package Manager: npm 6.14.13
OS: darwin x64

Angular: 12.0.0
... animations, cdk, cli, common, compiler, compiler-cli, core
... forms, material, platform-browser, platform-browser-dynamic
... router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1200.0
@angular-devkit/build-angular   12.0.0
@angular-devkit/core            12.0.0
@angular-devkit/schematics      12.0.0
@angular/flex-layout            11.0.0-beta.33
@schematics/angular             12.0.0
rxjs                            6.6.7
typescript                      4.2.4

</code></pre>

**Anything else relevant?**
Setting `"optimization": false` in the **build > options** builds the project file but the build budget size is 2 MB more.

```
"architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:browser",
          "options": {
            "outputPath": "dist/quiz-app",
            "optimization": false,
            ...
         }
     }
}
```

### PR DESCRIPTION
**Pull request by alan-agius4**
Workaround for Critters as it doesn't work when `@media all {}` is minified to `@media {}`.

Closes #20804

### CODE REVIEW

File: packages/angular_devkit/build_angular/src/browser/tests/options/inline-critical_spec.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+      expect(result?.success).toBe(true);
+      harness
+        .expectFile('dist/index.html')
+        .content.toContain(
+          `<link rel="stylesheet" href="styles.css" media="print" onload="this.media='all'">`,
+        );
+      harness.expectFile('dist/index.html').content.toContain(`body{color:#000;}`);
+    });
+
+    it(`should extract critical css when 'inlineCritical' is unset`, async () => {
```
Comment (clydin) [inline packages/angular_devkit/build_angular/src/browser/tests/options/inline-critical_spec.ts:43]:
```suggestion
    it(`should extract critical css when 'optimization' is unset`, async () => {
```
Should we add  a case for when `inlineCritical` itself is not set but `optimization` is set?

**Review by clydin — APPROVED** [review]

File: packages/angular_devkit/build_angular/src/browser/tests/options/inline-critical_spec.ts (hunk 1 above)
Comment (alan-agius4) [reply packages/angular_devkit/build_angular/src/browser/tests/options/inline-critical_spec.ts:43]: added

**Comment by filipesilva** [comment]
@alan-agius4 this PR does not apply cleanly to 12.1.x. Can you make a new PR targetting that branch please?

**Comment by alan-agius4** [comment]
Will do

**Pull request merged**